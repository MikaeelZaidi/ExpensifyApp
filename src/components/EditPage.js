import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense } from '../actions/expenses';
import { startRemoveExpense } from '../actions/expenses'

export class EditPage extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push("/")

    }

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push("/")
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>

                <div className="content-container">

                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />

                    <button onClick={this.onRemove} className="button button--secondary">Remove Expense</button>

                </div>

            </div>
        )
    }

}


const ConnectEditPage = (state, props) => {
    return {
        expense: state.expenses.find((exp) => {
            return exp.id === props.match.params.id
        })
    }
}

const mapDispatchtoProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

export default connect(ConnectEditPage, mapDispatchtoProps)(EditPage);