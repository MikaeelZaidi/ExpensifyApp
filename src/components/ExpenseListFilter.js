import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';
import { sortByAmount } from '../actions/filters';
import { sortByDate ,setStartDate,setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilter extends React.Component {

    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {

        this.props.dispatch(setStartDate(startDate))
        this.props.dispatch(setEndDate(endDate))
    };

    onFocusChange=(calendarFocused)=>{

        this.setState(()=>{
            calendarFocused
        })
    }

    render() {
        return (
            <div>
                <input type="text" defaultValue={this.props.filters.text} onChange={(e) => {

                    this.props.dispatch(setTextFilter(e.target.value))
                    console.log(e.target.value)
                }} />
                <select
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        if (e.target.value === 'amount') {
                            this.props.dispatch(sortByAmount())
                        }
                        else if (e.target.value === 'date') {
                            this.props.dispatch(sortByDate())
                        }
                    }}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>


                <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={()=>range}
                showClearDates={false}
                />
            </div>
        )
    }
}


const ExpenseListFilterConnect = connect((state) => {

    return {
        filters: state.filters
    }
})(ExpenseListFilter)

export default ExpenseListFilterConnect