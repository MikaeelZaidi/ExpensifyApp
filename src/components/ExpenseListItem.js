import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses'
import { NavLink } from 'react-router-dom';


const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => {

    return (
        <div>
            <NavLink to={`/edit/${id}`}> <h1>{description}</h1></NavLink>
            <p>{createdAt}-{amount}</p>
           
        </div>
    )

}

const ExpenseListItemConnect = connect()(ExpenseListItem)
export default ExpenseListItemConnect