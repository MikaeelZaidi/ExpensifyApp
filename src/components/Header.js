import React from 'react';
import { BrowserRouter, Route, Switch, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOut } from '../actions/auth'

export const Header = ({ startLogOut }) => {
    return (
        <header className="header">
            <div className="content-container">
                <div className="header__content">
                    <Link className="header__title" to="/dashboard" exact={true}>
                        <h1>Expensify</h1>
                    </Link>
                    <button className="button button--link" onClick={startLogOut} >Logout</button>
                </div>
            </div>

        </header>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatchToProps)(Header)