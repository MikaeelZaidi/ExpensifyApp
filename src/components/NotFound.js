import React from 'react';
import { BrowserRouter, Route, Switch, Link ,NavLink} from 'react-router-dom';

const NotFound = () => {
    return (
        <div>
            Yikes 404 <Link to="/">Go Home</Link>
        </div>
    )
}


export default NotFound