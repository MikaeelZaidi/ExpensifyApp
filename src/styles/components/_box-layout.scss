.box-layout{
    background:url('/images/TO.jpg');
    background-size:cover;
    height:100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    width:100vw

}

.box-layout__box{
    background:fade-out($color:rgb(226, 226, 247), $amount: 0.8);
    border-radius:4px;
    width:15rem;
    text-align: center;
    padding:$l-size $m-size;
}

.box-layout__title{
    margin:0
}
