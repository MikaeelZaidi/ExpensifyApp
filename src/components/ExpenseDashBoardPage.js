import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilter from './ExpenseListFilter';
import { connect } from 'react-redux';

const ExpenseDashBoardPage = (props) => {
    return (
        <div>
            <ExpenseListFilter />
            <ExpenseList />


        </div>
    )
}


export default ExpenseDashBoardPage