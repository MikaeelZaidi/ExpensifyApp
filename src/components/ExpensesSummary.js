import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import expensesSum from '../selectors/expense-total';
import ExpenseListItem from './ExpenseListItem';


export const ExpenseSummary = ({ expenseCount }) => {
    console.log(expenseCount)
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> with a total of </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>

            </div>

        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length
    }
};

export default connect(mapStateToProps)(ExpenseSummary);

