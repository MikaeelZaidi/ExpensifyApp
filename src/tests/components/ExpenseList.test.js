import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';
import {ExpenseList} from '../../components/ExpenseList';
import moment from 'moment';

const expenses = [{
    id: '1',
    description: 'Gum',
    note: "",
    amount: 192,
    createdAt: 0
}, {
    id: '2',
    description: 'Gas',
    note: "",
    amount: 1922,
    createdAt: moment(0).subtract(4, 'days').valueOf()
}, {
    id: '3',
    description: 'Rent',
    note: "",
    amount: 19211,
    createdAt: moment().add(3, 'days').valueOf()
}
]

test('This should test the expense list component', () => {

    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})


test('Should render expense list with empty message',()=>{
    const wrapper =shallow(<ExpenseList expenses={[]}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})