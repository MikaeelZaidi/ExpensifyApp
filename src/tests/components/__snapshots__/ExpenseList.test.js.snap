// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list with empty message 1`] = `
<div>
  <h1>
    Expense List:
  </h1>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`This should test the expense list component 1`] = `
<div>
  <h1>
    Expense List:
  </h1>
  <Connect(ExpenseListItem)
    amount={192}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <Connect(ExpenseListItem)
    amount={1922}
    createdAt={-345600000}
    description="Gas"
    id="2"
    key="2"
    note=""
  />
  <Connect(ExpenseListItem)
    amount={19211}
    createdAt={259200000}
    description="Rent"
    id="3"
    key="3"
    note=""
  />
</div>
`;
